File: Ontology/IObjectCardinality.cs
namespace Ontology
{
    public interface IObjectCardinality: IObjectPropertyRestriction
    {
        int              Cardinality     { get; }
        IClassExpression ClassExpression { get; }
    }

    public interface IObjectMinCardinality: IObjectCardinality
    {
    }

    public interface IObjectMaxCardinality: IObjectCardinality
    {
    }

    public interface IObjectExactCardinality: IObjectCardinality
    {
    }
}

File: Web/ClientApp/src/app/Ontology/IObjectCardinality.ts
import { ICardinality } from "./ICardinality";
import { IClassExpression } from "./IClassExpression";
import { IObjectPropertyRestriction } from "./IObjectPropertyRestriction";

export interface IObjectCardinality extends ICardinality, IObjectPropertyRestriction
{
    readonly ClassExpression: IClassExpression;
}

export interface IObjectMinCardinality extends IObjectCardinality
{
}

export interface IObjectMaxCardinality extends IObjectCardinality
{
}

export interface IObjectExactCardinality extends IObjectCardinality
{
}
